src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,248 @@
+	
+	/**
+	 * Accessing handlerMap hashmap from ContentCommunicator class to use it for sending message to UI
+	 */
+	private static HashMap<String, IContentNotificationHandler> handlerMap = ContentCommunicator.getHandlerMap();
+	
+	/**
+	 * Accessing imageMap hashmap from ContentCommunicator class to access image corresponding to username
+	 */
+	private static HashMap<String, String> imageMap = ContentCommunicator.getImageMap();
+	
+	/**
+	 * This variable will store the meta field of json string sent by the Networking module
+	 */
+	private static String metafield;
+	
+	/**
+	 * This array will store the jsonObject whose fields would be username and image
+	 */
+	private static JSONArray jsonObjectArray;
+	 
+	/**
+	 * This json object will store the elements of jsonObjectArray array
+	 */
+	private static JSONObject object;
+	
+	/**
+	 * This variable will store the username of client
+	 */
+	private static  String username;
+	
+	/**
+	 * This variable will store the image accessed by imageMap corresponding to username 
+	 */
+	private static String userimage;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board.
+	 * Both userName and userImage are of String type
+	 */
+	private static HashMap<String, String> imagemap;
+	
+	/**
+	* This variable will store a string which will contains log message that to be passed to log method of logger
+	*/
+	private static String logMessage;
+	
+	/**
+	 * Getting handler of the UI module to send the data
+	 */
+	private static IContentNotificationHandler handler;
+	 
+	/**
+	 * @param message - JSON string message (Meta fields are newUser or message or userExit)
+	 *
+	 * if meta field is newUser, then remaining field of parameter message would only be imageMap (it is a string of json array whose elements are json object having fields username and corresponding image). 
+	 *		onNewUserJoined(String message) - message fields are username and image  
+	 * if meta field is message, then remaining fields of parameter message would be username, message and time. onMessageReceived(String message) - message fields are username, message, time and image
+	 * if meta field is exitUser, then remaining field of parameter message would only be username. onUserExit(String message) - message fields are username and image
+	 * Networking module will be calling this method with json string message to send data to the content module
+	 *
+	 * creating error log message if message in not converting into json object,
+	 * or if not able extract field from object, or if not able to extract handler from hashmap (handlerMap)
+	 * if message will be successfully sent to UI, then creating success log message
+	 */
+	@Override
+	public void onMessageReceived(String message) {
+		logMessage = "onMessageReceived method of NetworkMessageHandler class is executing";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
```

Comment:
```suggestion
		logger.log(
		    ModuleID.INFRASTRUCTURE, 
		    LogLevel.INFO, 
		    "onMessageReceived method of NetworkMessageHandler class is executing"
		);
```
This is much cleaner. We don't need the `logMessage` variable.

Reply:
This needs to be changed in all places wherever you have logged.

Reply:
resolved